src/main/java/org/apache/mesos/state/SchedulerState.java

```diff
@@ -0,0 +1,135 @@
+package org.apache.mesos.state;
+
+import com.google.protobuf.TextFormat;
+import org.apache.mesos.Protos.*;
+import org.apache.mesos.curator.CuratorStateStore;
+import org.apache.mesos.offer.TaskException;
+import org.apache.mesos.offer.TaskUtils;
+import org.apache.mesos.reconciliation.TaskStatusProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+
+/**
+ * Read/write interface for the state of a Scheduler.
+ */
+public class SchedulerState implements TaskStatusProvider {
```

Comment:
I haven't gone through ever method, but I don't see this class adding a lot value, but it's adding some complexity, and doesn't really implement a clear interface, but selectively wraps a lot of what `StateStore` is doing.